### ISSUE TITLE
App is Damaged and Can't be opened (Old bug re-appearing?)

### ISSUE THREAD

**Issue by brendanmatkin** [processing/processing#4214]
I'm getting the "App is Damaged and Can't be opened" error. osx 10.11.1, Processing 3.0.1

Seems pretty much the same as issue #2095, but that was once fixed.

I can repeat it by zipping up and exported app, sending it to myself via email or dropbox, download & unzip. Try to run & get the damaged error. Changing OS security & privacy to "Allow apps downloaded from: anywhere" allows me to open it. 

Short version: I guess the apps are no longer being signed, or are being signed incorrectly? 

EDIT: Just noticed the 'code signing' note at the bottom of the export options dialog from Processing. I just searched the github issues when it happened and assumed this was the old bug popping up again. Anyway I see that it might be unfixable for now and will sign my apps manually. Thanks!

**Comment by benfry** [comment]
Yeah, that's why we have that ridiculously long block of text in the Export window. It's a headache on OS X, but we can't do much else.

**Comment by philipp-lehmann** [comment]
This solved it for me. Its only a fix if you want to install it yourself.

"Use xattr on the App Throwing the Damaged Error
This is sort of a last resort and is only recommended for advanced Mac users. Generally speaking if the app is still throwing a ‘damaged’ error message you might want to not use it. Use this at your own risk."

https://osxdaily.com/2019/02/13/fix-app-damaged-cant-be-opened-trash-error-mac/

**Comment by benfry** [comment]
Locking this issue because it's become a catch-all for unrelated issues with the same error message. If you have signing errors, please file a new issue. The issue originally posted about was resolved back in 2015/16.

**Comment by benfry** [comment]
For anyone finding this via search: there are (will be) additional code signing updates in the final release of Processing 4. You can also find more information in this just-posted wiki update: https://github.com/processing/processing4/wiki/Exporting-Applications#macos
